sdk/lusid_drive/utilities/utility_functions.py

```diff
@@ -29,3 +35,132 @@ def get_file_id(api_factory, file_name, folder_id):
     file_id = name_to_id(response, file_name)
 
     return file_id
+
+
+def create_folder(api_factory, folder_path, folder_name):
+    """
+    This function creates a new folder on LUSID Drive.
+
+    param api_factory ApiClientFactory: A LUSID Drive API Factory
+    param folder_path str: The folder path on LUSID Drive
+    param folder_path str: The new folder name on LUSID Drive
+
+    returns: a CreateFolder responses
+
+    """
+
+    folder_api = api_factory.build(lusid_drive.api.FoldersApi)
+
+    try:
+
+        create_folder_request = folder_api.create_folder(
+            create_folder=models.CreateFolder(folder_path, folder_name))
+
+        return create_folder_request
+
+    except lusid_drive.ApiException as e:
+
+        if json.loads(e.body)["code"] == 664:
+
+            return json.loads(e.body)["detail"]
+
+
+def create_nested_folder(api_factory, folder_path):
+    """
+    This function create a folder recursively.
+    For example, we can pass a new path "/a/b/c/d", and the function will recursively create the four API calls
+    to create:
+        /a
+        /a/b
+        /a/b/c
+        /a/b/c/d
+
+    :param api_factory ApiClientFactory: A LUSID Drive API Factory
+    :param folder_path str: The new folder path on LUSID Drive
+
+    returns: A list of CreateFolder responses
+    """
+
+    sub_dirs = [i for i in folder_path.split("/") if i != ""]
+
+    path = "/"
+
+    create_folder_requests = []
+
+    for elem in sub_dirs:
+
+        resp = create_folder(api_factory, path, elem)
+
+        create_folder_requests.append(resp)
+
+        path += elem + "/"
+
+    return create_folder_requests
+
+
+def path_to_drive_api_parms(drive_path):
+    """
+    Function to conver path of format /abc/def/text1 to a format suitable for the SearchApi
+
+    :param drive_path str: The path on LUSID Drive
+
+    returns: a tuple of (folder_path, folder_name). For example, the string "/abc/def/text1"
+    would return ("/abc/def", "text1")
+    """
+
+    if drive_path.count("/") == 1:
+
+        folder_path = "/"
+
+        f_name = drive_path[1:]
+
+    else:
+
+        folder_path = drive_path[:drive_path.rfind("/")]
+
+        f_name = drive_path[drive_path.rfind("/") + 1:]
+
+    return (folder_path, f_name)
+
+
+def delete_folder(api_factory, drive_path):
+    """
+    Function to delete folder on Drive
+
+    :param api_factory ApiClientFactory: A LUSID Drive API Factory
+    :param folder_path str: The folder path on LUSID Drive
+    """
+
+    if not drive_path.startswith("/"):
+        logger.info("The folder_path must start with a forward slash /")

```

Comment:
Do we want to fail this or add one in for them if it does not exist?

Reply:
Good point. It should fail. I have updated.